packages/backend-function/src/lambda-shims/resolve_ssm_params.ts

```diff
@@ -21,8 +19,16 @@ export const internalAmplifyFunctionResolveSsmParams = async (
     return;
   }
 
+  let actualSsmClient: SSM;
```

Comment:
should this var be in the module scope rather than the function scope so that subsequent invocations don't create a new client? We could name it something like `internalAmplifySsmClient` to reduce the liklihood of name collisions in the bundled file

Reply:
The old code seems to new-up a client for each call if I read it correctly, so this would be extra optimization that makes sense.
I'd however try to cache it without introducing new global symbol. I'll take a look.

Reply:
Yup this would be an improvement from the old code and sounds like a good idea to me. Maybe if we can't cache it then leave as-is to avoid new global symbol to reduce chances of name collisions?

Reply:
A summary of offline discussion.

1. We have determined that caching SSM client would be microoptimization at this point and we're punting this.
2. There are two options to cache it with pros and cons
    1. Hack around function prototype to cache it there. It requires type checking suppression and appears to be hacky. But it does not introduce new global symbol.
    2. Introduce new global symbol. Which is proper per TS compiler. But it increases likehood of collisions in customer code. We don't have strategy around prefixing/namespacing code we inject into lambdas so we should define it first.

Reply:
I've cut these:
https://github.com/aws-amplify/amplify-backend/issues/1524
https://github.com/aws-amplify/amplify-backend/issues/1525